src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -56,11 +60,11 @@ public EntityInteractHandler getItemHandler() {
 
                 p.playSound(p.getLocation(), Sound.ENTITY_ZOMBIE_VILLAGER_CONVERTED, 1, 1);
 
-                ZombieVillager zombieVillager = (ZombieVillager) entity;
-                zombieVillager.setConversionTime(1);
-
-                if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_15)) {
-                    zombieVillager.setConversionPlayer(p);
+                if (entity instanceof ZombieVillager) {
+                    healZombieVillager((ZombieVillager) entity, p);
+                }
+                else if (entity instanceof PigZombie) {
```

Comment:
Im aware that it is, but if there will ever be more recipes added, theyd have to create the else if statement back

Reply:
It will be their problem. Currently, this `else if` is redundant since we are working with only two cases (ZombieVillager and PigZombie), and the first one is already handled by an `if` check, for the second one it could simply be an `else`.

Reply:
I agree with @svr333 here, this `else if` clearly indicates what exact case this is for.
An else could lead to unwanted side effects if we ever add more entity types.

Reply:
Fair enough.